src/app/upgrade.cpp

```diff
@@ -228,11 +229,104 @@ namespace
+            {"BitTorrent/Session/AnonymousModeEnabled", "Preferences/Advanced/AnonymousMode"},
+            {"BitTorrent/Session/QueueingSystemEnabled", "Preferences/Queueing/QueueingEnabled"},
+            {"BitTorrent/Session/MaxActiveDownloads", "Preferences/Queueing/MaxActiveDownloads"},
+            {"BitTorrent/Session/MaxActiveUploads", "Preferences/Queueing/MaxActiveUploads"},
+            {"BitTorrent/Session/MaxActiveTorrents", "Preferences/Queueing/MaxActiveTorrents"},
+            {"BitTorrent/Session/IgnoreSlowTorrentsForQueueing", "Preferences/Queueing/IgnoreSlowTorrents"},
+            {"BitTorrent/Session/OutgoingPortsMin", "Preferences/Advanced/OutgoingPortsMin"},
+            {"BitTorrent/Session/OutgoingPortsMax", "Preferences/Advanced/OutgoingPortsMax"},
+            {"BitTorrent/Session/IgnoreLimitsOnLAN", "Preferences/Advanced/IgnoreLimitsLAN"},
+            {"BitTorrent/Session/IncludeOverheadInLimits", "Preferences/Advanced/IncludeOverhead"},
+            {"BitTorrent/Session/AnnounceIP", "Preferences/Connection/InetAddress"},
+            {"BitTorrent/Session/SuperSeedingEnabled", "Preferences/Advanced/SuperSeeding"},
+            {"BitTorrent/Session/MaxHalfOpenConnections", "Preferences/Connection/MaxHalfOpenConnec"},
+            {"BitTorrent/Session/uTPEnabled", "Preferences/Bittorrent/uTP"},
+            {"BitTorrent/Session/uTPRateLimited", "Preferences/Bittorrent/uTP_rate_limited"},
+            {"BitTorrent/TrackerEnabled", "Preferences/Advanced/trackerEnabled"},
+            {"Network/Proxy/OnlyForTorrents", "Preferences/Connection/ProxyOnlyForTorrents"},
+            {"Network/Proxy/Type", "Preferences/Connection/ProxyType"},
+            {"Network/Proxy/Authentication", "Preferences/Connection/Proxy/Authentication"},
+            {"Network/Proxy/Username", "Preferences/Connection/Proxy/Username"},
+            {"Network/Proxy/Password", "Preferences/Connection/Proxy/Password"},
+            {"Network/Proxy/IP", "Preferences/Connection/Proxy/IP"},
+            {"Network/Proxy/Port", "Preferences/Connection/Proxy/Port"},
+            {"Network/PortForwardingEnabled", "Preferences/Connection/UPnP"},
+            {"AddNewTorrentDialog/TreeHeaderState", "AddNewTorrentDialog/qt5/treeHeaderState"},
+            {"AddNewTorrentDialog/Width", "AddNewTorrentDialog/width"},
+            {"AddNewTorrentDialog/Position", "AddNewTorrentDialog/y"},
+            {"AddNewTorrentDialog/Expanded", "AddNewTorrentDialog/expanded"},
+            {"AddNewTorrentDialog/SavePathHistory", "TorrentAdditionDlg/save_path_history"},
+            {"AddNewTorrentDialog/Enabled", "Preferences/Downloads/NewAdditionDialog"},
+            {"AddNewTorrentDialog/TopLevel", "Preferences/Downloads/NewAdditionDialogFront"},
+
+            {"State/BannedIPs", "Preferences/IPFilter/BannedIPs"}
+        };
+
+        auto *settingsStorage = SettingsStorage::instance();
+        for (const QPair<QString, QString> &pair : keyMapping)
```

Comment:
Couldn't work the following?
```c++
        for (const auto [newKey, oldKey] : keyMapping)
```
I would prefer to see meaningful names instead of `pair.first`/`pair.second`.

Reply:
I took another approach as it seems to be better to me.